### PR TITLE
CI: build the rust tools during CI

### LINKED ISSUE #1076: Nothing tests Rust libraries in /tools
Need something in testing infrastructure to build these at minimum so that they stay up to date as rust versions move

### PR DESCRIPTION
**Pull request by ppannuto**
### Pull Request Overview

This just adds the stuff that can be compiled in tools/ to the travis build

This would also run their tests if they had any

Closes #1076.

### Testing Strategy

Compiling.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,6 +45,10 @@ ci-travis:
 	@printf "$$(tput bold)******************$$(tput sgr0)\n"
 	@CI=true ./tools/run_cargo_fmt.sh diff
 	@./tools/check_wildcard_imports.sh
+	@printf "$$(tput bold)*************$$(tput sgr0)\n"
+	@printf "$$(tput bold)* CI: Tools *$$(tput sgr0)\n"
+	@printf "$$(tput bold)*************$$(tput sgr0)\n"
+	@for f in `./tools/list_tools.sh`; do echo "$$(tput bold)Build & Test $$f"; pushd tools/$$f > /dev/null; CI=true RUSTFLAGS="-D warnings" cargo build --all-targets || exit 1; popd > /dev/null; done
```
Comment (alevy) [inline Makefile]: I believe this seems to be failing, as it seems to not to actually go into the tool directory before running cargo.

**Comment by ppannuto** [comment]
Well, that turned out to be not such an easy box, but fixed all the travis issues now
